### PR TITLE
IPFS filenames

### PR DESCRIPTION
**Pull request by callumhyland-dc**
This PR adds support for filename preservation by utilising the `wrap-with-directory` flag from IPFS

### CODE REVIEW

**Review by jonmattgray — APPROVED** [review]

File: app/util/appUtil.js [hunk 1]
```diff
@@ -47,35 +48,58 @@ api.on('error', (err) => {
...
-const downloadFile = async (hash) => {
-  const url = `http://${IPFS_HOST}:${IPFS_PORT}/api/v0/cat?arg=${hash}`
-  const res = await fetch(url, { method: 'POST' })
-
-  if (!res.ok) {
-    throw new Error(`Error fetching file from IPFS (${res.status}): ${await res.text()}`)
-  }
-
-  return res.body
+const downloadFile = async (dirHash) => {
```
Comment (mattdean-digicatapult) [inline app/util/appUtil.js:82]: Hang on does this deal with the old case where we just store the file and not the directory

**Review by mattdean-digicatapult — COMMENTED** [review]
Couple of questions

File: app/util/appUtil.js [hunk 2]
```diff
@@ -47,35 +48,58 @@ api.on('error', (err) => {
...
 
 async function processMetadata(file) {
   if (file) {
-    const response = await addFile(file)
-    if (response && response.Hash && response.Name && response.Size) {
-      const decoded = bs58.decode(response.Hash)
+    const responses = await addFile(file)
+
+    // directory has no Name
+    const dir = responses.find((r) => r.Name === '')
```
[the remark this answers is not in the document]
Comment (callumhyland-dc) [reply app/util/appUtil.js:72]: Finding the correct directory hash (which has no `Name` field, unlike the file, which does have a name)

File: app/util/appUtil.js (hunk 1 above)
Comment (callumhyland-dc) [reply app/util/appUtil.js:82]: Not yet!

**Review by mattdean-digicatapult — APPROVED** [review]

**Pull request merged**